### PR TITLE
Implement `__geo_interface__` for indicator and report classes

### PR DESCRIPTION
**Pull request by matthiasschaub**
### Description

See PR #66

### CODE REVIEW

File: workers/tests/integrationtests/test_geo_interface.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+class TestGeoInterface(unittest.TestCase):
+    def test(self):
+        feature = asyncio.run(
+            db_client.get_region_from_db(feature_id=3, fid_field="ogc_fid")
+        )
+        indicator = GhsPopComparisonBuildings(
+            feature=feature, layer_name="building_count"
+        )
+        print(indicator.feature)
```
Comment (matthiasschaub) [inline workers/tests/integrationtests/test_geo_interface.py:21]: This works as expected.

File: workers/tests/integrationtests/test_geo_interface.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        print(indicator.__geo_interface__)
```
Comment (matthiasschaub) [inline workers/tests/integrationtests/test_geo_interface.py:22]: Traceback (most recent call last):
  File "/home/matthias/work/oqt/workers/tests/integrationtests/test_geo_interface.py", line 22, in test
    print(indicator.__geo_interface__)
  File "/home/matthias/work/oqt/workers/ohsome_quality_analyst/base/indicator.py", line 108, in __geo_interface__
    "geometry": self.feature.geometry,
AttributeError: 'GhsPopComparisonBuildings' object has no attribute 'feature'

**Comment by matthiasschaub** [comment]
```bash
cd workers/tests/integrationtests
python test_geo_interface.py
{"geometry": {"coordinates": [[[[8.573519, 49.415844], [8.573125, 49.418096], [8.573602, 49.419967], [8.573137, 49.423648], [8.57776, 49.42532], [8.579572, 49.426242], [8.582888, 49.426469], [8.592864, 49.428812], [8.611134, 49.425455], [8.628308, 49.442962], [8.637725, 49.440277], [8.647106, 49.437602], [8.707905, 49.446383], [8.716206, 49.451763], [8.724684, 49.448596], [8.743338, 49.446934], [8.745999, 49.45398], [8.741644, 49.457488], [8.752892, 49.459332], [8.767943, 49.454273], [8.762008, 49.449545], [8.766739, 49.438141], [8.788416, 49.431047], [8.789941, 49.416687], [8.793216, 49.41046], [8.791873, 49.403912], [8.787115, 49.401839], [8.777999, 49.401612], [8.773679, 49.390183], [8.769462, 49.387665], [8.763519, 49.387265], [8.767844, 49.379545], [8.75502, 49.377409], [8.752547, 49.376444], [8.750183, 49.375517], [8.749349, 49.373812], [8.740263, 49.374336], [8.73117, 49.372358], [8.725257, 49.37439], [8.725259, 49.371116], [8.729116, 49.36789], [8.725838, 49.360442], [8.722815, 49.357364], [8.712037, 49.35558], [8.67493, 49.360645], [8.672847, 49.358114], [8.665655, 49.357579], [8.66451, 49.354379], [8.662033, 49.35426], [8.651996, 49.353727], [8.657231, 49.361494], [8.646111, 49.369879], [8.639986, 49.368719], [8.637725, 49.362532], [8.634015, 49.352379], [8.622614, 49.353352], [8.622976, 49.359493], [8.6024, 49.366157], [8.607315, 49.376967], [8.612846, 49.378247], [8.617598, 49.379362], [8.6324, 49.38678], [8.637725, 49.386869], [8.641426, 49.38693], [8.646667, 49.389265], [8.643391, 49.392889], [8.637725, 49.393259], [8.635895, 49.393737], [8.630921, 49.412538], [8.611559, 49.40539], [8.601979, 49.409237], [8.577429, 49.412229], [8.573519, 49.415844]]]], "type": "MultiPolygon"}, "properties": {}, "type": "Feature"}
E
======================================================================
ERROR: test (__main__.TestGeoInterface)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/matthias/work/oqt/workers/tests/integrationtests/test_geo_interface.py", line 22, in test
    print(indicator.__geo_interface__)
  File "/home/matthias/work/oqt/workers/ohsome_quality_analyst/base/indicator.py", line 108, in __geo_interface__
    "geometry": self.feature.geometry,
AttributeError: 'GhsPopComparisonBuildings' object has no attribute 'feature'

----------------------------------------------------------------------
Ran 1 test in 0.201s

FAILED (errors=1)
```

**Comment by joker234** [comment]
> @joker234 I am out of my depth here on why following error gets raised. If you could take a look at this issue, I would appreciate it very much. I am happy to talk you through this after our meeting on Monday.
> 
> ```shell
> cd workers/tests/integrationtests
> python test_geo_interface.py
> {"geometry": {"coordinates": [[[[8.573519, 49.415844], [8.573125, 49.418096], [8.573602, 49.419967], [8.573137, 49.423648], [8.57776, 49.42532], [8.579572, 49.426242], [8.582888, 49.426469], [8.592864, 49.428812], [8.611134, 49.425455], [8.628308, 49.442962], [8.637725, 49.440277], [8.647106, 49.437602], [8.707905, 49.446383], [8.716206, 49.451763], [8.724684, 49.448596], [8.743338, 49.446934], [8.745999, 49.45398], [8.741644, 49.457488], [8.752892, 49.459332], [8.767943, 49.454273], [8.762008, 49.449545], [8.766739, 49.438141], [8.788416, 49.431047], [8.789941, 49.416687], [8.793216, 49.41046], [8.791873, 49.403912], [8.787115, 49.401839], [8.777999, 49.401612], [8.773679, 49.390183], [8.769462, 49.387665], [8.763519, 49.387265], [8.767844, 49.379545], [8.75502, 49.377409], [8.752547, 49.376444], [8.750183, 49.375517], [8.749349, 49.373812], [8.740263, 49.374336], [8.73117, 49.372358], [8.725257, 49.37439], [8.725259, 49.371116], [8.729116, 49.36789], [8.725838, 49.360442], [8.722815, 49.357364], [8.712037, 49.35558], [8.67493, 49.360645], [8.672847, 49.358114], [8.665655, 49.357579], [8.66451, 49.354379], [8.662033, 49.35426], [8.651996, 49.353727], [8.657231, 49.361494], [8.646111, 49.369879], [8.639986, 49.368719], [8.637725, 49.362532], [8.634015, 49.352379], [8.622614, 49.353352], [8.622976, 49.359493], [8.6024, 49.366157], [8.607315, 49.376967], [8.612846, 49.378247], [8.617598, 49.379362], [8.6324, 49.38678], [8.637725, 49.386869], [8.641426, 49.38693], [8.646667, 49.389265], [8.643391, 49.392889], [8.637725, 49.393259], [8.635895, 49.393737], [8.630921, 49.412538], [8.611559, 49.40539], [8.601979, 49.409237], [8.577429, 49.412229], [8.573519, 49.415844]]]], "type": "MultiPolygon"}, "properties": {}, "type": "Feature"}
> E
> ======================================================================
> ERROR: test (__main__.TestGeoInterface)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "work/oqt/workers/tests/integrationtests/test_geo_interface.py", line 22, in test
>     print(indicator.__geo_interface__)
>   File "work/oqt/workers/ohsome_quality_analyst/base/indicator.py", line 108, in __geo_interface__
>     "geometry": self.feature.geometry,
> AttributeError: 'GhsPopComparisonBuildings' object has no attribute 'feature'
> 
> ----------------------------------------------------------------------
> Ran 1 test in 0.201s
> 
> FAILED (errors=1)
> ```

`vars(…)` returns a **reference** to the attribute dict (`__dict__`) of an object. If you use `something.pop(…)`, you'll change the actual object. I would suggest to use `vars(…).copy()` instead.

**Comment by matthiasschaub** [comment]
Close in favor of #86

**Pull request closed without merging**